runtime-config.md

```diff
@@ -12,27 +12,28 @@ Presently there are `Prestart`, `Poststart` and `Poststop`.
 Hooks allow one to run code before/after various lifecycle events of the container.
 Hooks MUST be called in the listed order.
 The state of the container is passed to the hooks over stdin, so the hooks could get the information they need to do their work.
+All hooks execute in the host environment (e.g. the same namespace, cgroups, etc. that apply to the host process).
 
 Hook paths are absolute and are executed from the host's filesystem.
 
 ### Pre-start
 
-The pre-start hooks are called after the container process is spawned, but before the user supplied command is executed.
+The pre-start hooks are called [after the container process is spawned, but before the user supplied command is executed](runtime.md#typical-lifecycle).
 They are called after the container namespaces are created on Linux, so they provide an opportunity to customize the container.
 In Linux, for e.g., the network namespace could be configured in this hook.
 
-If a hook returns a non-zero exit code, then an error including the exit code and the stderr is returned to the caller and the container is torn down.
+If a hook returns a non-zero exit code, [then an error including the exit code and the stderr is returned to the caller and the container is torn down](runtime.md#typical-lifecycle).
 
 ### Post-start
 
-The post-start hooks are called after the user process is started.
+The post-start hooks are called [after the user process is started](runtime.md#typical-lifecycle).
```

Comment:
On Mon, Oct 12, 2015 at 04:03:30PM -0700, Vish Kannan wrote:
> Not really. If you bind-mount the ns files, you don't need a process
> to hold the namespaces.

Oh, that will make my life a lot easier :).  Is that:

Host process                    Temporary child
--------------------------      -----------------------------------------
Launch a child process      ->
                                Call clone(2) with CLONE_NEW*
                                Bind mount /proc/self/ns/* somewhere else
                                pivot_root, etc.
                            <-  Exit
Use the bind-mounted files

In that case, excepting the PID namespace from the pre-start
requirements seems like a reasonable prices to pay.  And it will make
explicit ‘create’ and ‘start’ commands possible [1].

On the other hand, we'll probably need to revisit the state JSON that
landed in #87 now that the ‘pid’ entry doesn't make sense for *either*
pre-start or post-stop.

[1]: https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/frUXLljXy8Y
     Message-ID: <20150915034717.GO18018@odin.tremily.us>